### PR TITLE
Unifies energy gun self charging

### PR DESCRIPTION
**Pull request by Jordie0608**
Unsnowflakes duplicated charging and shot type selection of various energy guns. This allows for any energy gun to be set to self charge. All cyborg guns now recharge from the cyborg's cell after each shot rather than passively.

### CODE REVIEW

**Comment by WJohn** [comment]
Neat :+1:

**Comment by Nitrousoxide** [comment]
Presumably this would also fix this issue:
https://github.com/tgstation/-tg-station/issues/10469

File: code/modules/projectiles/guns/energy.dm [hunk 1]
```diff
@@ -33,9 +36,29 @@
 	shot = ammo_type[select]
 	fire_sound = shot.fire_sound
 	fire_delay = shot.delay
+	SSobj.processing |= src
```
Comment (JJRcop) [inline code/modules/projectiles/guns/energy.dm]: Hmmm... Wish there was a less footprinty way to do this while allowing admins to make guns self-charge via VV.

Comment (Anonus) [inline code/modules/projectiles/guns/energy.dm]: This adds it to proccessing objects even if it's not self charging, which means all you have to do is edit selfcharge to 1 to get it to recharge.

Comment (duncathan) [inline code/modules/projectiles/guns/energy.dm]: >adds it to processing even if it's not self charging

Oh god no that's a horrible idea

Comment (Iamgoofball) [inline code/modules/projectiles/guns/energy.dm]: if only we had a proc that gets called on var updates

**Comment by Aranclanos** [comment]
cmon don't add all energy guns to the object process list, that's stupid
I really like the removal of copypaste, but work on the issue

**Comment by Jordie0608** [comment]
I'm open to better ideas but all I've got is "remove the spontaneous selfcharge by varediting" which would at worst be a bit boring. Meanwhile it's not exactly an expensive process() although I can't say for sure how it'd stack up.

**Comment by Jordie0608** [comment]
Will be updating this to use on_varedit proc linked above whenever that gets merged.

**Comment by Jordie0608** [comment]
This is ready now

**Comment by Anonus** [comment]
I don't see where the gun is added to processing on New(), only where it's added when varedited.

**Comment by Jordie0608** [comment]
That was the handling prior to the existence of the on_varedit proc

**Comment by Anonus** [comment]
Does New() really call on_varedit for its vars? Huh.

**Comment by Jordie0608** [comment]
Oh I misunderstood what you meant, my bad

**Pull request merged**